pkg/virt-controller/watch/migration/migration.go

```diff
@@ -2084,17 +2079,18 @@ func (c *Controller) deleteVMI(obj interface{}) {
 	}
 }
 
-func (c *Controller) outboundMigrationsOnNode(node string, runningMigrations []*virtv1.VirtualMachineInstanceMigration) (int, error) {
+func (c *Controller) outboundMigrationsOnNode(node string, runningMigrations []*virtv1.VirtualMachineInstanceMigration) int {
 	sum := 0
 	for _, migration := range runningMigrations {
 		key := controller.NamespacedKey(migration.Namespace, migration.Spec.VMIName)
-		if vmi, exists, _ := c.vmiStore.GetByKey(key); exists {
-			if vmi.(*virtv1.VirtualMachineInstance).Status.NodeName == node {
-				sum = sum + 1
+		if obj, exists, _ := c.vmiStore.GetByKey(key); exists {
```

Comment:
We need to check the error here, otherwise it will lead to a wrong calculation.

Reply:
I'm also not a fan of relying on cached state for another resource here and agree this implementation needs to be stripped out but for the sake of the release I think the approach taken by the PR is good enough for now.